### ISSUE TITLE
buffer overflow suspected by asan

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#1261]
build:
https://github.com/haproxy/haproxy/runs/2605305752

```
***  h1    debug|==11382==ERROR: AddressSanitizer: global-buffer-overflow on address 0x00000148b740 at pc 0x00000051c844 bp 0x7ffca46e28a0 sp 0x7ffca46e2898
  ***  h1    debug|READ of size 2 at 0x00000148b740 thread T0
  **** dT    0.055
  ***  h1    debug|    #0 0x51c843 in eb_next_dup /home/runner/work/haproxy/haproxy/include/import/ebtree.h:619:47
  ***  h1    debug|    #1 0x51b735 in ebmb_next_dup /home/runner/work/haproxy/haproxy/include/import/ebmbtree.h:83:9
  ***  h1    debug|    #2 0x51b70e in ssl_store_get_cafile_entry /home/runner/work/haproxy/haproxy/src/ssl_ckch.c:975:8
  **** dT    0.056
  ***  h1    debug|    #3 0x509d97 in ssl_set_cert_crl_file /home/runner/work/haproxy/haproxy/src/ssl_sock.c:323:30
  **** dT    0.057
  ***  h1    debug|    #4 0x5099c3 in ssl_set_verify_locations_file /home/runner/work/haproxy/haproxy/src/ssl_sock.c:352:9
  ***  h1    debug|    #5 0x4fa4f8 in ssl_sock_prepare_ctx /home/runner/work/haproxy/haproxy/src/ssl_sock.c:4285:20
  ***  h1    debug|    #6 0x4fa067 in ssl_sock_prep_ctx_and_inst /home/runner/work/haproxy/haproxy/src/ssl_sock.c:4466:13
  ***  h1    debug|    #7 0x4fc65a in ssl_sock_prepare_all_ctx /home/runner/work/haproxy/haproxy/src/ssl_sock.c:4905:14
  ***  h1    debug|    #8 0x4fcad6 in ssl_sock_prepare_bind_conf /home/runner/work/haproxy/haproxy/src/ssl_sock.c:4992:9
  **** dT    0.058
  ***  h1    debug|    #9 0x72dea3 in check_config_validity /home/runner/work/haproxy/haproxy/src/cfgparse.c:4011:8
  ***  h1    debug|    #10 0x7da8ce in init /home/runner/work/haproxy/haproxy/src/haproxy.c:1936:14
  ***  h1    debug|    #11 0x7d79c9 in main /home/runner/work/haproxy/haproxy/src/haproxy.c:2849:2
  **** dT    0.061
  ***  h1    debug|    #12 0x7fcb4ccf90b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
  ***  h1    debug|    #13 0x43cfcd in _start (/home/runner/work/haproxy/haproxy/haproxy+0x43cfcd)
  ***  h1    debug|
  ***  h1    debug|0x00000148b740 is located 32 bytes to the left of global variable 'ckchs_transaction.0' defined in 'src/ssl_ckch.c' (0x148b760) of size 8
  ***  h1    debug|0x00000148b740 is located 16 bytes to the right of global variable 'cafile_tree' defined in 'src/ssl_ckch.c:951:16' (0x148b720) of size 16
  ***  h1    debug|SUMMARY: AddressSanitizer: global-buffer-overflow /home/runner/work/haproxy/haproxy/include/import/ebtree.h:619:47 in eb_next_dup
  ***  h1    debug|Shadow bytes around the buggy address:
  ***  h1    debug|  0x000080289690: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896a0: f9 f9 f9 f9 04 f9 f9 f9 f9 f9 f9 f9 00 00 f9 f9
  ***  h1    debug|  0x0000802896b0: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896c0: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896d0: f9 f9 f9 f9 00 00 00 f9 f9 f9 f9 f9 01 f9 f9 f9
  ***  h1    debug|=>0x0000802896e0: f9 f9 f9 f9 00 00 f9 f9[f9]f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896f0: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289700: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289710: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289720: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289730: f9 f9 f9 f9 00 00 00 00 00 f9 f9 f9 f9 f9 f9 f9
  ***  h1    debug|Shadow byte legend (one shadow byte represents 8 application bytes):
  ***  h1    debug|  Addressable:           00
  ***  h1    debug|  Partially addressable: 01 02 03 04 05 06 07 
  ***  h1    debug|  Heap left redzone:       fa
  ***  h1    debug|  Freed heap region:       fd
  ***  h1    debug|  Stack left redzone:      f1
  ***  h1    debug|  Stack mid redzone:       f2
  ***  h1    debug|  Stack right redzone:     f3
  ***  h1    debug|  Stack after return:      f5
  ***  h1    debug|  Stack use after scope:   f8
  ***  h1    debug|  Global redzone:          f9
  ***  h1    debug|  Global init order:       f6
  ***  h1    debug|  Poisoned by user:        f7
  ***  h1    debug|  Container overflow:      fc
  ***  h1    debug|  Array cookie:            ac
  ***  h1    debug|  Intra object redzone:    bb
  ***  h1    debug|  ASan internal:           fe
  ***  h1    debug|  Left alloca redzone:     ca
  ***  h1    debug|  Right alloca redzone:    cb
  ***  h1    debug|  Shadow gap:              cc
  ***  h1    debug|==11382==ABORTING
```

```
***  h1    debug|==11058==ERROR: AddressSanitizer: global-buffer-overflow on address 0x00000148b740 at pc 0x00000051c844 bp 0x7ffe18717dd0 sp 0x7ffe18717dc8
  ***  h1    debug|READ of size 2 at 0x00000148b740 thread T0
  **** dT    0.228
  ***  h1    debug|    #0 0x51c843 in eb_next_dup /home/runner/work/haproxy/haproxy/include/import/ebtree.h:619:47
  ***  h1    debug|    #1 0x51b735 in ebmb_next_dup /home/runner/work/haproxy/haproxy/include/import/ebmbtree.h:83:9
  ***  h1    debug|    #2 0x51b70e in ssl_store_get_cafile_entry /home/runner/work/haproxy/haproxy/src/ssl_ckch.c:975:8
  ***  h1    debug|    #3 0x51b779 in ssl_store_get0_locations_file /home/runner/work/haproxy/haproxy/src/ssl_ckch.c:987:30
  ***  h1    debug|    #4 0x51bd84 in ssl_store_load_locations_file /home/runner/work/haproxy/haproxy/src/ssl_ckch.c:1091:22
  ***  h1    debug|    #5 0x5292d0 in srv_parse_ca_file /home/runner/work/haproxy/haproxy/src/cfgparse-ssl.c:1339:7
  **** dT    0.230
  ***  h1    debug|    #6 0x6a1856 in _srv_parse_kw /home/runner/work/haproxy/haproxy/src/server.c:2559:13
  ***  h1    debug|    #7 0x6a04b6 in parse_server /home/runner/work/haproxy/haproxy/src/server.c:2682:14
  **** dT    0.232
  ***  h1    debug|    #8 0x7c0070 in cfg_parse_listen /home/runner/work/haproxy/haproxy/src/cfgparse-listen.c:348:15
  **** dT    0.233
  ***  h1    debug|    #9 0x723656 in readcfgfile /home/runner/work/haproxy/haproxy/src/cfgparse.c:2285:13
  ***  h1    debug|    #10 0x7da659 in init /home/runner/work/haproxy/haproxy/src/haproxy.c:1810:10
  ***  h1    debug|    #11 0x7d79c9 in main /home/runner/work/haproxy/haproxy/src/haproxy.c:2849:2
  **** dT    0.236
  ***  h1    debug|    #12 0x7f3de303e0b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
  ***  h1    debug|    #13 0x43cfcd in _start (/home/runner/work/haproxy/haproxy/haproxy+0x43cfcd)
  ***  h1    debug|
  **** dT    0.237
  ***  h1    debug|0x00000148b740 is located 32 bytes to the left of global variable 'ckchs_transaction.0' defined in 'src/ssl_ckch.c' (0x148b760) of size 8
  ***  h1    debug|0x00000148b740 is located 16 bytes to the right of global variable 'cafile_tree' defined in 'src/ssl_ckch.c:951:16' (0x148b720) of size 16
  ***  h1    debug|SUMMARY: AddressSanitizer: global-buffer-overflow /home/runner/work/haproxy/haproxy/include/import/ebtree.h:619:47 in eb_next_dup
  ***  h1    debug|Shadow bytes around the buggy address:
  ***  h1    debug|  0x000080289690: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896a0: f9 f9 f9 f9 04 f9 f9 f9 f9 f9 f9 f9 00 00 f9 f9
  ***  h1    debug|  0x0000802896b0: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896c0: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896d0: f9 f9 f9 f9 00 00 00 f9 f9 f9 f9 f9 01 f9 f9 f9
  ***  h1    debug|=>0x0000802896e0: f9 f9 f9 f9 00 00 f9 f9[f9]f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x0000802896f0: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289700: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289710: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289720: f9 f9 f9 f9 00 f9 f9 f9 f9 f9 f9 f9 00 f9 f9 f9
  ***  h1    debug|  0x000080289730: f9 f9 f9 f9 00 00 00 00 00 f9 f9 f9 f9 f9 f9 f9
  ***  h1    debug|Shadow byte legend (one shadow byte represents 8 application bytes):
  ***  h1    debug|  Addressable:           00
  ***  h1    debug|  Partially addressable: 01 02 03 04 05 06 07 
  ***  h1    debug|  Heap left redzone:       fa
  ***  h1    debug|  Freed heap region:       fd
  ***  h1    debug|  Stack left redzone:      f1
  ***  h1    debug|  Stack mid redzone:       f2
  ***  h1    debug|  Stack right redzone:     f3
  ***  h1    debug|  Stack after return:      f5
  ***  h1    debug|  Stack use after scope:   f8
  ***  h1    debug|  Global redzone:          f9
  ***  h1    debug|  Global init order:       f6
  ***  h1    debug|  Poisoned by user:        f7
  ***  h1    debug|  Container overflow:      fc
  ***  h1    debug|  Array cookie:            ac
  ***  h1    debug|  Intra object redzone:    bb
  ***  h1    debug|  ASan internal:           fe
  ***  h1    debug|  Left alloca redzone:     ca
  ***  h1    debug|  Right alloca redzone:    cb
  ***  h1    debug|  Shadow gap:              cc
  ***  h1    debug|==11058==ABORTING
````

**Comment by wlallemand** [comment]
Thanks, we are already looking into it.

**Comment by rlebreton** [comment]
This should be fixed on master now, thanks to commit 2608e348bec8a533225424fad06760ce6e19f167 BUG/MEDIUM: ebtree: Invalid read when looking for dup entry

**Comment by chipitsine** [comment]
it is marked for backporting. I'm keeping this issue open
